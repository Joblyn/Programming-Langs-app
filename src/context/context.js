import { createContext } from "react";

export const RouteContext = createContext(null);

export const NameContext = createContext(null);