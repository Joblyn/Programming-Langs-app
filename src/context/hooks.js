export const useAppContext = (value = null) => {
  let route = value;
  return route;
};