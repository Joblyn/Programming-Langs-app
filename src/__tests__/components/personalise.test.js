import React from "react";
import { render, cleanup } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { AppContext } from "../../context/context";
import Personalise from "../../components/personalise/personalise";

describe("<Personalise />", () => {
  afterEach(cleanup);

  it("renders component correctly", () => {
    const name = "Joblyn";
    const { container, getByAltText } = render(
      <AppContext.Provider value={{ name }}>
        <Personalise />
      </AppContext.Provider>
    );
    expect(getByAltText("_three bottles_")).toBeTruthy();
    expect(container.getElementsByTagName("p")[0].innerHTML).toEqual("Personalising for Joblyn");
  });
});
