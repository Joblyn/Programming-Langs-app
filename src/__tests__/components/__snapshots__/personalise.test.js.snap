// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Personalise /> renders component correctly 1`] = `
<div
  class="h-full pt-10 w-full"
>
  <div
    class="mb-4"
  >
    <img
      alt="_three bottles_"
      class="h-36 w-auto max-w-32"
      src="personalise.avif"
    />
  </div>
  <div
    class="tracking-wider"
  >
    <p
      class="text-lg font-bold tracking-wider"
    >
      Personalising for 
      Joblyn
    </p>
  </div>
</div>
`;
